### ISSUE TITLE
Reset State of Core Functions

### ISSUE THREAD

**Issue by einfallstoll** [synesenom/ran#19]
My knowledge in this field is very limited, but I try to give you an example of what I want to achieve.

I have the following code to seed the core and shuffle an array:
```
const ran = require('ranjs');

const values = [1, 5, 6, 7, 2, 4];

for (var i = 0; i < 10; i++) {
  ran.core.seed(1);
  console.log(ran.core.shuffle(values));
}
```

... and I receive the following output (it's always the same because of the seeding):
```
[ 1, 7, 5, 2, 6, 4 ]
[ 1, 2, 7, 6, 5, 4 ]
[ 1, 6, 2, 5, 7, 4 ]
[ 1, 5, 6, 7, 2, 4 ]
[ 1, 7, 5, 2, 6, 4 ]
[ 1, 2, 7, 6, 5, 4 ]
[ 1, 6, 2, 5, 7, 4 ]
[ 1, 5, 6, 7, 2, 4 ]
[ 1, 7, 5, 2, 6, 4 ]
[ 1, 2, 7, 6, 5, 4 ]
```

Which is expected, however, I want the output to be this:
```
[ 1, 7, 5, 2, 6, 4 ]
[ 1, 7, 5, 2, 6, 4 ]
[ 1, 7, 5, 2, 6, 4 ]
[ 1, 7, 5, 2, 6, 4 ]
[ 1, 7, 5, 2, 6, 4 ]
[ 1, 7, 5, 2, 6, 4 ]
[ 1, 7, 5, 2, 6, 4 ]
[ 1, 7, 5, 2, 6, 4 ]
[ 1, 7, 5, 2, 6, 4 ]
[ 1, 7, 5, 2, 6, 4 ]
```

I want an array to be always shuffled the same based on the seed I provide. Is this possible somehow?

**Comment by synesenom** [comment]
Hej Fabio,

Thanks for your question! You see a different array every time in your for loop because the `shuffle` method shuffles the arrays in-place (maybe I should make this more emphasized in the docs). So the shuffling with the same seed works as you expect if you pass the *same array of values* each time (instead of the variable `values` which gets updated at every shuffle):

```
const ran = require('ranjs');

for (var i = 0; i < 10; i++) {
  ran.core.seed(1);
  // Here I pass an array with the same values each time
  console.log(ran.core.shuffle([1, 5, 6, 7, 2, 4]));
}
```

I hope this helps 🙂

**Comment by einfallstoll** [comment]
OMG. I didn't even think of this! I'm used to the fact that if the function does return an array it's a copy and not shuffled in-place.

I will test this as soon as I'm home.

**Comment by einfallstoll** [comment]
Ok, I tested it and can confirm it works. I just compared it so `Array.sort()` and it will sort the array in-place *and* returns it, so I would prefer not to alter the behaviour but to mention this in the docs.

The fun thing about this is, after this library I checked out a bunch of other libraries all having the same issues and guess what ... it's because I copied the `shuffle` function from your library into my code for testing purposes. 🤦
